### PR TITLE
do not create instance group with NEG backends

### PR DESCRIPTION
**Pull request by freehan**
ref: https://github.com/kubernetes/ingress-gce/issues/433

### CODE REVIEW

File: pkg/controller/utils.go [hunk 1]
```diff
@@ -86,3 +86,15 @@ func convert(ings []*extensions.Ingress) (retVal []interface{}) {
 	}
 	return
 }
+
+// getNodePortsUsedByIngress returns the list of uniq NodePort from the input ServicePorts.
+// Only NonNEG service backend need NodePort.
+func getNodePortsUsedByIngress(svcPorts []utils.ServicePort) []int64 {
```
Comment (rramkumar1) [inline pkg/controller/utils.go]: Can we call this nodePorts()?

**Comment by lbernail** [comment]
Thanks! I can test it on our clusters as soon as this is merged to give you feedback

**Comment by freehan** [comment]
> Aside from the PR, I'm thinking it might be worthwhile to add an e2e test for this in the main k8s repo. Maybe even bake the check into an existing test you have.

That requires those tests to be serialized though.

**Pull request merged**